src/Payum/Core/Security/AbstractTokenFactory.php

```diff
@@ -53,10 +54,12 @@ public function createToken($gatewayName, $model, $targetPath, array $targetPara
         }
 
         if ($afterPath && 0 === strpos($afterPath, 'http')) {
-            $afterUrl = Url::createFromUrl($afterPath);
-            $afterUrl->getQuery()->modify($afterParameters);
+            $afterUri = HttpUri::createFromString($afterPath);
+
+            $modifier = new MergeQuery(Query::createFromArray($afterParameters)->__toString());
+            $afterUri = $modifier->__invoke($afterUri);
```

Comment:
I am also fine with something like this `call_user_func($modifier, $afterUri);`. Explicit call of magic method `__invoke` looks weird  to me.